Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -149,11 +153,31 @@ public struct RequestResponseOperationOptions: CStructWithUserData, Sendable {
 
     typealias RawType = aws_mqtt_request_operation_options
     func withCStruct<Result>(userData: UnsafeMutableRawPointer?, _ body: (RawType) -> Result) -> Result {
-        // TODO: convert into aws_mqtt_request_operation_options
-        let options = aws_mqtt_request_operation_options()
-        return body(options)
+        var raw_options = aws_mqtt_request_operation_options()
+        return self.subscriptionTopicFilters.withMutableByteCursorArray { topicsCursor, len in
+            raw_options.subscription_topic_filters = topicsCursor
```

Comment:
This creation of the CStruct is pretty different from what we're doing for unsubscribe packets. Specifically the use of this func: https://github.com/awslabs/aws-crt-swift/blob/011a801f3c01aea854df5f01308222644c3778bb/Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift#L465 to store the topic filter strings. Does this mean we're overcomplicating things for the unsubscribe packet? We should stick to a single pattern if possible. Right now I'm not entirely sure why we decided to make a copy of the [String] for use as an aws_byte_cursor but I feel like we probably had a reason. Wish we left better comments >,<

Reply:
Nice catch. We should be able to removed the `convertTopicFilters()` function and use the same pattern for unsubscribe packet.